src/main/java/net/minecraftforge/oredict/ShapelessOreRecipe.java

```diff
@@ -123,6 +125,41 @@ public boolean matches(@Nonnull InventoryCrafting inv, @Nonnull World world)
         return RecipeMatcher.findMatches(items, this.input) != null;
     }
 
+    @Override
+    public NonNullList<ItemStack> getRemainingItems(InventoryCrafting inv)
+    {
+        NonNullList<ItemStack> remainder = NonNullList.withSize(inv.getSizeInventory(), ItemStack.EMPTY);
+        NonNullList<ItemStack> items = NonNullList.withSize(inv.getSizeInventory(), ItemStack.EMPTY);
+        for (int index = 0; index < inv.getSizeInventory(); index++)
+        {
+            items.set(index, inv.getStackInSlot(index));
+        }
+        int[] mapping = RecipeMatcher.findMatches(items, this.input);
```

Comment:
Call this based on whether this is simple.

Reply:
Hes also not using it correctly.

Reply:
What did I missed?

Reply:
https://github.com/MinecraftForge/MinecraftForge/blob/89db87dbfce4e4bd130f5daccac7cac495002275/patches/minecraft/net/minecraft/item/crafting/ShapelessRecipes.java.patch#L57
You need to only put in valid items.
You also nee to make sure the incoming arrays are equal in length to your inputs.